app/src/ui/changes/changes-list.tsx

```diff
@@ -567,6 +567,31 @@ export class ChangesList extends React.Component<
         action: () => this.props.onIgnoreFile(path),
         enabled: Path.basename(path) !== GitIgnoreFileName,
       })
+
+      const pathComponents = path.split(Path.sep);
+      const assembledPaths = []
+      for (let i = 0; i < pathComponents.length - 1; i++) {
+        // Add each parent directory to the list of paths to ignore
+        // Also include a / since it's the absolute compared to the git repository
+        // That way, we prevent git from ignoring the wrong directory which has the same name
+        // but is in a different location
+        assembledPaths.push('/' + pathComponents.slice(0, i + 1).join(Path.sep) + '/')
+      }
+
+      if (pathComponents.length > 0) {
+        items.push({
+          label: __DARWIN__
+          ? 'Ignore Folder (Add to .gitignore)'
+          : 'Ignore folder (add to .gitignore)',
+          submenu: assembledPaths.map((assembledPath) => ({
```

Comment:
A bit of a nit, but for better efficiency/less code, we do not need two loops here and the `if(pathComponents.lenght > 0)` could be right after line 571. 

The `for` loop we could go ahead and construct the submenu items instead of only the assembled paths. 

Something like:
``` 
if(pathComponents.length > 0) {
   const submenu = []
   for(...) {
         const assembledPath = ...;
         submenu.push({label: assembledPath, action: ...(assembledPath)})
    }
    items.push({label: ..., submenu, enabled: ..);
}
```

Reply:
Patched up fully!